flink-formats/flink-parquet/src/test/java/org/apache/flink/formats/parquet/ParquetFilesystemStatisticsReportTest.java

```diff
@@ -0,0 +1,242 @@
+                                + " c * 3.14 as g, h, i, j"
+                                + " from originalT")
+                .await();
+        assertThat(Objects.requireNonNull(new File(path1).listFiles()).length).isEqualTo(2);
+
+        FlinkStatistic statistic =
+                getStatisticsFromOptimizedPlan("select * from parquetFileSystemTable");
+
+        Map<String, ColumnStats> expectedColumnStatsMap = new HashMap<>();
+        expectedColumnStatsMap.put("a", new ColumnStats.Builder().setNullCount(2L).build());
+        expectedColumnStatsMap.put(
+                "b", new ColumnStats.Builder().setMax(3).setMin(1).setNullCount(0L).build());
+        expectedColumnStatsMap.put(
+                "c", new ColumnStats.Builder().setMax(2).setMin(-4).setNullCount(0L).build());
+        expectedColumnStatsMap.put(
+                "d", new ColumnStats.Builder().setMax(3L).setMin(-5L).setNullCount(0L).build());
+        expectedColumnStatsMap.put(
+                "e", new ColumnStats.Builder().setMax(90.08).setMin(-0.8).setNullCount(0L).build());
+        expectedColumnStatsMap.put(
+                "f",
+                new ColumnStats.Builder().setMax("e fg").setMin("abcd").setNullCount(2L).build());
+        expectedColumnStatsMap.put(
+                "g",
+                new ColumnStats.Builder()
+                        .setMax(BigDecimal.valueOf(6.28))
+                        .setMin(BigDecimal.valueOf(-12.56))
+                        .setNullCount(0L)
+                        .build());
+        expectedColumnStatsMap.put(
+                "h",
+                new ColumnStats.Builder()
+                        .setMax(Date.valueOf("2017-12-12"))
+                        .setMin(Date.valueOf("2017-12-12"))
+                        .setNullCount(2L)
+                        .build());
+        expectedColumnStatsMap.put(
+                "i",
+                new ColumnStats.Builder()
+                        .setMax(Time.valueOf("10:08:09"))
+                        .setMin(Time.valueOf("10:08:09"))
+                        .setNullCount(0L)
+                        .build());
+        // Now parquet store timestamp as type int96, and int96 now not support statistics, so
+        // timestamp not support statistics now.
+        expectedColumnStatsMap.put("j", new ColumnStats.Builder().setNullCount(0L).build());
+        int expectedRowCount = 6;
+
+        assertThat(statistic.getTableStats())
+                .isEqualTo(new TableStats(expectedRowCount, expectedColumnStatsMap));
+    }
+
+    private FlinkStatistic getStatisticsFromOptimizedPlan(String sql) {
+        RelNode relNode = TableTestUtil.toRelNode(tEnv.sqlQuery(sql));
+        RelNode optimized = util.getPlanner().optimize(relNode);
+        FlinkStatisticVisitor visitor = new FlinkStatisticVisitor();
+        visitor.go(optimized);
+        return visitor.result;
+    }
+
+    private static class FlinkStatisticVisitor extends RelVisitor {
```

Comment:
The class `FlinkStatisticVisitor` appears to be repeated in several test classes for support multiple format reporting statistic. Can you avoid repeating yourself by introducing a public test parent class or extracting this class separately?

Reply:
> The class `FlinkStatisticVisitor` appears to be repeated in several test classes for support multiple format reporting statistic. Can you avoid repeating yourself by introducing a public test parent class or extracting this class separately?

I will abstract these test methods and test class relationships when implement hive source stats report in this issue: https://issues.apache.org/jira/browse/FLINK-27988. The coding work has been completed, and I will push this pr soon.